src/main/java/scrolls/elder/logic/commands/RedoCommand.java

```diff
@@ -0,0 +1,25 @@
+package scrolls.elder.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+
+import scrolls.elder.logic.commands.exceptions.CommandException;
+import scrolls.elder.model.Model;
+
+/**
+ * Reverses the effect of a previous undo operation on the Datastore.
+ */
+public class RedoCommand extends Command {
+    public static final String COMMAND_WORD = "redo";
+    public static final String MESSAGE_SUCCESS = "Previous undo operation has been reversed!";
+    public static final String MESSAGE_REDO_ERROR = "No previous undo operation to be reversed";
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        if (!model.getDatastoreVersionStorage().canRedo()) {
```

Comment:
Checking if there is a previous operation to be redone is important, but how might that tie in with the assert statement in redoChanges() in terms of 2 layers of checking?

Reply:
I treated the assert statement in redoChanges more like a sanity check to ensure correctness, although it might be fine to remove the assertion as well